packager/packager.cc

```diff
@@ -1435,11 +1416,20 @@ ast::ParsedFile rewritePackagedFile(core::Context ctx, ast::ParsedFile parsedFil
 
 // Re-write source files to be in packages. This is only called if no package definitions were
 // changed.
-vector<ast::ParsedFile> rewritePackagedFilesFast(core::GlobalState &gs, vector<ast::ParsedFile> files) {
-    Timer timeit(gs.tracer(), "packager.rewritePackagedFilesFast");
+vector<ast::ParsedFile> rewriteFilesFast(core::GlobalState &gs, vector<ast::ParsedFile> files) {
+    Timer timeit(gs.tracer(), "packager.rewriteFilesFast");
     for (auto i = 0; i < files.size(); i++) {
         core::Context ctx(gs, core::Symbols::root(), files[i].file);
-        files[i] = rewritePackagedFile(ctx, move(files[i]));
+        if (files[i].file.data(gs).isPackage()) {
+            {
+                core::MutableContext ctx(gs, core::Symbols::root(), files[i].file);
+                // Re-write imports and exports:
+                getPackageInfo(ctx, files[i], gs.packageDB().extraPackageFilesDirectoryPrefixes());
```

Comment:
I verified that, in `pipeline.cc`, we don't unfreeze the name table or symbol table for incremental packager so debug builds will enforce that this does not define any new names.